src/java/htsjdk/samtools/CRAMFileReader.java

```diff
@@ -562,64 +495,46 @@ public SAMRecordIterator assertSorted(final SortOrder sortOrder) {
 
         @Override
         public void close() {
-            delegate.close();
+            if (unfilteredIterator != null) {
+                unfilteredIterator.close();
+            }
         }
 
         @Override
         public boolean hasNext() {
```

Comment:
Not a fan of `hasNext()` methods that have side effects like this -- ideally the next record would be preloaded in the constructor and then updated in `next()`. If this is too hard to change, though, we can keep this as-is.